spec/controllers/vm_infra_controller_spec.rb

```diff
@@ -328,6 +328,43 @@
     expect(response.status).to eq(200)
   end
 
+  context 'transform VM dialog' do
+    let(:dialog)          { double("Dialog") }
+    let(:resource_action) { double("ResourceAction") }
```

Comment:
You should use `FactoryGirl.create` instead of using test doubles. There's already one for the `Dialog`, however, you'd need to write your own for the `ResourceAction`.

Reply:
@skateman : can you show @smelamud, where to put the factories (I guess spec/factories) ? We have no factories in ui-classic yet (all are in the core repo after the repo split).

Reply:
@martinpovolny factories are model-related, therefore, they live in the core repo. For example the one for dialogs is here: https://github.com/ManageIQ/manageiq/blob/master/spec/factories/dialog.rb

I don't think you'd need more complicated than this for the `ResourceAction`:
```ruby
FactoryGirl.define do
  factory :resource_action do
  end
end
```

Then in your tests you can do something like this:
```
let(:dialog) { FactoryGirl.create(:dialog) }
let(:resource_action) { FactoryGirl.create(:resource_action, :dialog => dialog)
``

Reply:
@skateman : we cannot ask people to place factories needed ONLY for UI code into the core repo.

If we need a factory that is not already present in the core repo then we need to create it in the UI repo.

Reply:
@martinpovolny okay, but what if someone later wants to use the factory elsewhere? For example in the core repo?

Reply:
What if... someone later ;-) Ok, someone later may need to move the factory or create a new one. But I don't want to further complicate the not very popular task of writing tests by making people do PRs to 2 repositories. Where in one they would get a "pending core" label while in the other they would be asked "why is this needed". It just does not make any sense.

Reply:
Well the best case scenario would be to ask the core guys to write (at least empty) factories (as the one above) when they specify new models, but I get your point.